radiant/filters.cpp

```diff
@@ -259,6 +283,51 @@ bool FilterBrush( brush_t *pb ){
 						break;
 					}
 				}
+				// idTech2 material filters
+				else if ( filters->attribute == 8 ) {
+					switch ( filters->exclude ) {
+						case EXCLUDE_LIQUIDS:
+							if ( f->texdef.contents & (CONTENTS_WATER | CONTENTS_LAVA | CONTENTS_SLIME ) ) {
+								filterbrush = true;
+							}
+							if ( strstr( f->pShader->getName(), "water" ) ||
+								 strstr( f->pShader->getName(), "lava" )  ||
+								 strstr( f->pShader->getName(), "slime" ) ) {
+								filterbrush = true;
+							}
+							break;
+						case EXCLUDE_HINTSSKIPS:
+							if ( f->texdef.flags & (SURF_HINT | SURF_SKIP) ) {
+								filterbrush = true;
+							}
+							if ( strstr( f->pShader->getName(), "hint" ) ||
+								 strstr( f->pShader->getName(), "skip" ) ) {
+								filterbrush = true;
+							}
+							break;
+						case EXCLUDE_TRANSLUCENT:
+							if ( f->texdef.contents & CONTENTS_WINDOW ) {
+								filterbrush = true;
+							}
+							if ( f->texdef.flags & (SURF_TRANS33 | SURF_TRANS66 | SURF_ALPHA_TEST) ) {
+								filterbrush = true;
+							}
+							if ( strstr( f->pShader->getName(), "glass" ) ||
+								 strstr( f->pShader->getName(), "window" ) ) {
+								filterbrush = true;
+							}
+							break;
+						case EXCLUDE_SKY:
+							if ( f->texdef.flags & SURF_SKY ) {
+								filterbrush = true;
+							}
+							if ( strstr( f->pShader->getName(), "sky" ) ) {
+								filterbrush = true;
+							}
+							break;
+
+					}
+				}
```

Comment:
This is the meat of the patch. It makes the stock GtkRadiant "exclude" filters work with Quake and Quake2 derivatives.